code/obj/item/clothing/chameleon_js.dm

```diff
@@ -269,3 +272,1659 @@
+	desc = "A knit cap in red."
+	icon_state = "red"
+	item_state = "swat_gl"
+	wear_image_icon = 'icons/mob/clothing/head.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_headgear.dmi'
+	icon = 'icons/obj/clothing/item_hats.dmi'
+	uses_multiple_icon_states = 1
+	var/list/clothing_choices = list()
+	blocked_from_petasusaphilic = TRUE
+	item_function_flags = IMMUNE_TO_ACID
+	seal_hair = 0
+
+	New()
+		..()
+		for(var/U in (typesof(/datum/chameleon_hat_pattern)))
+			var/datum/chameleon_hat_pattern/P = new U
+			src.clothing_choices += P
+		return
+
+	attackby(obj/item/clothing/head/U as obj, mob/user as mob)
+		if(istype(U, /obj/item/clothing/head/chameleon))
+			boutput(user, "<span class='alert'>No!!! That's a terrible idea! You'll cause a cataclysmic hat infinite loop!</span>")
+			SPAWN_DBG(1 SECOND)
+				boutput(user, "<span class='alert'>Nah, just yankin' your chain. Doing that still doesn't work though!</span>")
+			return
+
+		if(istype(U, /obj/item/clothing/head/))
+			for(var/datum/chameleon_hat_pattern/P in src.clothing_choices)
+				if(P.name == U.name)
+					boutput(user, "<span class='alert'>That appearance is already saved in the chameleon pattern banks!</span>")
+					return
+
+			var/datum/chameleon_hat_pattern/P = new /datum/chameleon_hat_pattern(src)
+			P.name = U.name
+			P.desc = U.desc
+			P.icon_state = U.icon_state
+			P.item_state = U.item_state
+			P.sprite_item = U.icon
+			P.sprite_worn = U.wear_image_icon
+			P.sprite_hand = U.inhand_image_icon
+			P.seal_hair = U.seal_hair
+			src.clothing_choices += P
+
+			boutput(user, "<span class='notice'>[U.name]'s appearance has been copied!</span>")
+
+	emp_act()
+		if (ishuman(src.loc))
+			var/mob/living/carbon/human/M = src.loc
+			boutput(M, "<span class='alert'><B>Your chameleon hat malfunctions!</B></span>")
+			src.name = "hat"
+			src.desc = "A knit cap in...what the hell?"
+			wear_image = image(wear_image_icon)
+			inhand_image = image(inhand_image_icon)
+			src.icon_state = "psyche"
+			src.item_state = "bgloves"
+			src.seal_hair = 0
+			M.set_clothing_icon_dirty()
+
+	verb/change()
+		set name = "Change Appearance"
+		set desc = "Alter the appearance of your Chameleon Hat."
+		set category = "Local"
+		set src in usr
+
+		var/datum/chameleon_hat_pattern/which = input("Change the hat to which pattern?", "Chameleon Hat") as null|anything in clothing_choices
+
+		if(!which)
+			return
+
+		src.change_outfit(which)
+
+	proc/change_outfit(var/datum/chameleon_hat_pattern/T)
+		src.name = T.name
+		src.desc = T.desc
+		src.icon_state = T.icon_state
+		src.item_state = T.item_state
+		src.icon = T.sprite_item
+		src.wear_image_icon = T.sprite_worn
+		src.inhand_image_icon = T.sprite_hand
+		src.wear_image = image(wear_image_icon)
+		src.inhand_image = image(inhand_image_icon)
+		src.seal_hair = T.seal_hair
+		usr.set_clothing_icon_dirty()
+
+/datum/chameleon_hat_pattern
+	var/name = "hat"
+	var/desc = "A knit cap in red."
+	var/icon_state = "red"
+	var/item_state = "red"
```

Comment:
should probably also be `rgloves` assuming this is supposed to look identical to the red beanie